basyx.submodelrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/TestSubmodelRepositoryHTTP.java

```diff
@@ -98,24 +99,60 @@ public void getSpecificSubmodelNonExisting() throws IOException {
 	@Test
 	public void updateExistingSubmodel() throws IOException, ParseException {
 		String id = "7A7104BDAB57E184";
-		String expectedSubmodelJSON = getSubmodelUpdateJSON();
+		String expectedSubmodelJSON = getUpdatedSubmodelJSON();
 
 		CloseableHttpResponse creationResponse = putSubmodel(id, expectedSubmodelJSON);
 
 		assertEquals(HttpStatus.NO_CONTENT.value(), creationResponse.getCode());
 
-		String submodelJSON = getSpecificSubmodelJSON(id);
+		String submodelJSON = requestSpecificSubmodelJSON(id);
 		assertSameJSONContent(expectedSubmodelJSON, submodelJSON);
 	}
 
 	@Test
 	public void updateNonExistingSubmodel() throws IOException {
 		String id = "nonExisting";
-		String expectedSubmodelJSON = getSubmodelUpdateJSON();
+		String expectedSubmodelJSON = getUpdatedSubmodelJSON();
 
-		CloseableHttpResponse creationResponse = putSubmodel(id, expectedSubmodelJSON);
+		CloseableHttpResponse updateResponse = putSubmodel(id, expectedSubmodelJSON);
+
+		assertEquals(HttpStatus.NOT_FOUND.value(), updateResponse.getCode());
+	}
+
+	@Test
+	public void createSubmodelNewId() throws IOException, ParseException {
+		String submodelJSON = getNewSubmodelJSON();
+		CloseableHttpResponse creationResponse = createSubmodel(submodelJSON);
+
+		assertSubmodelCreationReponse(submodelJSON, creationResponse);
+
+		String requestedSubmodel = requestSpecificSubmodelJSON("newSubmodel");
```

Comment:
To be sure that the id of requestedSubmodel _is_ the same ID as the one used in submodelJson, it would be appropriate to extract that id from submodelJSON and use it e.g. like this `requestSpecificSubmodelJSON("extractedId")`. This ensures that the method `getNewSubmodelJSON()` resp. `SingleSubmodelNew.json` could be changed without necessarily touching the Test.